core/src/main/scala/org/apache/spark/h2o/converters/SparkDataFrameConverter.scala

```diff
@@ -166,10 +167,12 @@ private[h2o] object SparkDataFrameConverter extends Logging {
           case _: ml.linalg.VectorUDT => con.putVector(idxH2O, row.getAs[ml.linalg.Vector](idxField), elemSizes(idxField))
           case _: mllib.linalg.VectorUDT => con.putVector(idxH2O, row.getAs[mllib.linalg.Vector](idxField), elemSizes(idxField))
           case udt: UserDefinedType[_] => throw new UnsupportedOperationException(s"User defined type is not supported: ${udt.getClass}")
-          case _ => con.putNA(idxH2O, idxField)
+          case unsupported => throw new UnsupportedOperationException(s"Data of type ${unsupported.getClass} are not supported for the conversion" +
```

Comment:
I believe we should throw exception if we bump to type we don't understand, instead of silently passing NA

Reply:
Yeah, it's better to fail fast then to get unexpected results.